spring-data-relational/src/main/java/org/springframework/data/relational/core/sql/SelectBuilder.java

```diff
@@ -500,6 +500,17 @@
 		 * @see Table#column(String)
 		 */
 		SelectOnConditionComparison on(Expression column);
+
+		/**
+		 * Declare a join condition in one step.
+		 *
+		 * This is harder to use but more flexible then the fluent style of {@link #on(Expression)} which only allows for equality comparisons chained together with `AND`.
```

Comment:
How about: 

> Using conditions allows more flexibility in comparison to  {@link #on(Expression)} which only allows for equality comparisons chained together with `AND`.

Harder to use has the appeal of letting users shying away from such a method.